packages/editor/src/lib/editor/Editor.ts

```diff
@@ -7251,8 +7276,21 @@ export class Editor extends EventEmitter<TLEventMap> {
 			const shape = this.getShape(partial.id)
 			if (!shape) continue
 
-			// If the shape is locked and we're not setting isLocked to true, continue
-			if (this.isShapeOrAncestorLocked(shape) && !Object.hasOwn(partial, 'isLocked')) continue
+			// If we're "forcing" the update, then we'll update the shape
+			// regardless of whether it / its ancestor is locked
+			if (!this._force) {
+				if (shape.isLocked) {
+					// If the shape itself is locked (even if one of its ancestors is
+					// also locked) then only allow an update that unlocks the shape.
+					if (!(Object.hasOwn(partial, 'isLocked') && !partial.isLocked)) {
+						continue
+					}
+				} else if (this.isShapeOrAncestorLocked(shape)) {
+					// If the shape itself is unlocked, and any of the shape's
+					// ancestors are locked then we'll skip the update
+					continue
+				}
+			}
```

Comment:
Previously, you could update a locked shape just by having the `isLocked` as part of the update. A call like `updateShape({...myLockedShape, x: 100 })` would go through because the `myLockedShape` would have the `isLocked` property on it. From here on, if you want to update a locked shape for any reason (apart from unlocking it) you'll need to use `force`.

Likewise (though less important) you could update a shape with a locked ancestor just by including `isLocked` in the partial, even if `isLocked` would produce no change.